lib/wda/webdriveragent.js

```diff
@@ -155,10 +155,16 @@ class WebDriverAgent {
     }
   }
 
-  async uninstall () {
-    log.debug(`Removing WDA application from device`);
+  /**
+   * Remove WDA from the test device.
+   *
+   * @param {?string} bundleId The bundle id which will be removed from test device.
+   */
+  async uninstall (bundleId = null) {
+    const removeBundleId = util.hasValue(bundleId) ? bundleId : WDA_BUNDLE_ID;
```

Comment:
is WDA_RUNNER_BUNDLE_ID equal to WDA_BUNDLE_ID ?

Reply:
No.
WDA_RUNNER_BUNDLE_ID => 'com.facebook.WebDriverAgentRunner'
WDA_BUNDLE_ID => 'com.apple.test.WebDriverAgentRunner-Runner'

`WDA_BUNDLE_ID` is used when a user build WDA to a simulator or a real device with personal provisioning profile while WDA's bundle Id is `WDA_RUNNER_BUNDLE_ID`
Only when users use their enterprise provisioning profile for example, we must provide `updatedWDABundleId` and respect it when we uninstall it.

Reply:
I actually prefer throwing exceptions in these cases. I like defensive coding more than dummy proof coding